### DISCUSSION TITLE
Adding dependencies to the rpath of the binary?

### DISCUSSION THREAD

**Discussion by expenses** [ipetkov/crane#717] [Q&A] [answered]
Hi, I'm attempting to use crane to build a bevy app. Unfortunately, it doesn't work as is (see https://github.com/bevyengine/bevy/blob/4bf647ff3b0ca7c8ca47496db9cfe03702328473/docs/linux_dependencies.md#nix) and requires a few extra libraries to be specified. The following wrapper works:
```nix
   runnable = pkgs.writeShellApplication {
            name = "runnable";
            text = "${default}/bin/repl-gui";
            runtimeEnv.LD_LIBRARY_PATH =
              with pkgs;
              lib.makeLibraryPath [
                libxkbcommon
                vulkan-loader
              ];
          };
```
I want to be able to directly patch the binary in order to avoid needing a wrapper script. Ideally adding this would work:
```nix
fixupPhase = ''
                patchelf --add-rpath ${
                  with pkgs; lib.makeLibraryPath [
                    libxkbcommon
                    vulkan-loader
                  ]
                } $out/bin/repl-gui
              '';
```
At the moment this gets run for the deps derivation though which is definitely not what we want:
```
       > Running phase: fixupPhase
       > patchelf: getting info about '/nix/store/rim753r2ivl6x8xykgyyjnih1fv4gqyp-repl-gui-deps-0.1.0/bin/repl-gui': No such file or directory
       For full logs, run 'nix log /nix/store/wvrpwzfna5irqwfzmb64gfv7kv3y8dmw-repl-gui-deps-0.1.0.drv'.
error: 1 dependencies of derivation '/nix/store/4jcbpqsyl8nyhvgrhwza10ykb0r29c03-repl-gui-0.1.0.drv' failed to build
```

**Comment by ipetkov** [comment, marked as answer]
Hi @expenses ! I just added a FAQ entry about this in https://github.com/ipetkov/crane/pull/722. Please take a look at the following and let me know if it does not answer your question!

https://github.com/ipetkov/crane/blob/c052ed5e849b75d15875b58ae326df45f07ee7a4/docs/faq/control-when-hooks-run.md

**Reply by expenses** [reply]
Oh yeah, of course that's the answer, duh! Thanks very much!

**Reply by hustlerone** [reply]
I'm an absolute noob i literally tried this and it don't work for me

**Reply by hustlerone** [reply]
nvm got it thanks for the magical post
